### DISCUSSION TITLE
Changing DrawVideo TextureName before start

### DISCUSSION THREAD

**Discussion by orena1** [bonsai-rx/bonsai#866] [Q&A] [answered]
Hi, I am trying to run a simple script that plays videos randomly, I have modified a bit this script from @glopesdev:
https://groups.google.com/g/bonsai-users/c/AyNzAkWENnc/m/fIh0r2-NAQAJ

The workflow is very simple, I have a PythonSource that in this case just goes over the videos one by one (for this example I iterate over the videos for simplicity):
![image](https://user-images.githubusercontent.com/8983713/161832589-ebe24f02-7da1-46c7-adc0-1fdbd27f66af.png)

The main problem is that once I start the Workflow the first thing that happens is that the TextureName that was already present in the DrawVideo properties is played and only after that then one that is supplied from my PythonSource.

As you can see in the workflow I have tried to use ReplaySubject as  Gonçalo suggested it to so to change the CSVWriter filename before its start, so I thought it might help, but it did not work.

Is there any way to do it?
I've added the workflow and a zip with the videos:
[first_video.zip](https://github.com/bonsai-rx/bonsai/files/8421244/first_video.zip)
[first_video.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/8421245/first_video.bonsai.zip)

Thanks.

**Comment by bruno-f-cruz** [comment, marked as answer]
I refactored the workflow a bit but from your description i think it works for your needs. Let me know if something is not clear.

[first_video.zip](https://github.com/bonsai-rx/bonsai/files/8440960/first_video.zip)

**Reply by orena1** [reply]
Got it, thanks!
